gensim/models/word2vec.py

```diff
@@ -725,31 +731,43 @@ def worker_loop():
         push_done = False
         done_jobs = 0
         next_alpha = self.alpha
-        jobs_source = enumerate(utils.grouper(sentences, chunksize))
+
+        job_batch = []
+        batch_size = 0
+        MAX_SENTENCE_LEN = 200  # TODO: consider proper value for this constant.
+        job_no = 0
         # fill jobs queue with (sentence, alpha) job tuples
-        while True:
-            try:
-                job_no, items = next(jobs_source)
-                logger.debug("putting job #%i in the queue at alpha %.05f", job_no, next_alpha)
-                job_queue.put((items, next_alpha))
-                # update the learning rate before every next job
-                if self.min_alpha < next_alpha:
-                    if total_examples:
-                        # examples-based decay
-                        pushed_examples += len(items)
-                        next_alpha = self.alpha - (self.alpha - self.min_alpha) * (pushed_examples / total_examples)
-                    else:
-                        # words-based decay
-                        pushed_words += self._raw_word_count(items)
-                        next_alpha = self.alpha - (self.alpha - self.min_alpha) * (pushed_words / total_words)
-                    next_alpha = max(next_alpha, self.min_alpha)
-            except StopIteration:
-                logger.info(
-                    "reached end of input; waiting to finish %i outstanding jobs",
-                    job_no - done_jobs + 1)
-                for _ in xrange(self.workers):
-                    job_queue.put((None, 0))  # give the workers heads up that they can finish -- no more work!
-                push_done = True
+        for sent in sentences:
+            if batch_size + len(sent) < MAX_SENTENCE_LEN:
```

Comment:
misnomer; this is not the maximum length of a sentence, but rather the maximum number of words in a batch (as @gojomo pointed out)